File: BasicPrograms/MostCommonWord.py
import re
class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        pattern = r'[^a-zA-Z0-9\s]'
        cleaned_paragraph = re.sub(pattern, ' ', paragraph.lower())
        words = cleaned_paragraph.replace(".","").split()
        word_count = {}
        for word in words:
            if word not in banned:
                if word in word_count:
                    word_count[word] += 1
                else:
                    word_count[word] = 1
        most_freq_word = None
        most_freq_count = 0
        for word, count in word_count.items():
            if count > most_freq_count:
                most_freq_word = word
                most_freq_count = count

        return most_freq_word

File: BasicPrograms/MostCommonWord.java
import java.util.*;

class Solution {
    public String mostCommonWord(String paragraph, String[] banned) {
        Set<String> bannedSet = new HashSet<>(Arrays.asList(banned));
        String[] words = paragraph.toLowerCase().split("[^a-zA-Z]+");
        Map<String, Integer> wordCount = new HashMap<>();
        for (String word : words) {
            if (!bannedSet.contains(word)) {
                wordCount.put(word, wordCount.getOrDefault(word, 0) + 1);
            }
        }
        String mostCommonWord = "";
        int mostCommonCount = 0;
        for (Map.Entry<String, Integer> entry : wordCount.entrySet()) {
            if (entry.getValue() > mostCommonCount) {
                mostCommonWord = entry.getKey();
                mostCommonCount = entry.getValue();
            }
        }
        
        return mostCommonWord;
    }
}
